### DISCUSSION TITLE
Playing encrypted widevine streams?

### DISCUSSION THREAD

**Discussion by shilangyu** [jhomlala/betterplayer#544] [Q&A] [answered]
Hello, has anyone managed to get `betterplayer` working with encrypted widevine streams? I tried every combination of the DRM config but none of them worked. I would expect it to look something of that sort:

```dart
BetterPlayerDrmConfiguration(
   drmType: BetterPlayerDrmType.widevine,
   licenseUrl: 'https://foo.bar',
   token: 'Bearer=my.token.here',
)
```

However I had no success. Perhaps this is a `betterplayer` limitation (not supported?). I am using Azure Media Services with flags `(format=mpd-time-csf,encryption=cenc)`. Any help would be appreciated, thanks.

PS. 
I might add that the length of the stream is correctly fetched and displayed, but the playback plays no content

**Comment by jhomlala** [comment, marked as answer]
You should use `headers` instead of `token` for Widevine DRM. So you should modify your code:
```dart
...
headers: {"Authorization":"Bearer <token>"}
...
```
